plugins/catalog-backend/src/next/database/DefaultProcessingDatabase.ts

```diff
@@ -355,8 +355,16 @@ export class DefaultProcessingDatabase implements ProcessingDatabase {
   ): Promise<GetProcessableEntitiesResult> {
     const tx = txOpaque as Knex.Transaction;
 
-    const items = await tx<DbRefreshStateRow>('refresh_state')
-      .select()
+    let itemsQuery = tx<DbRefreshStateRow>('refresh_state').select();
+
+    // This avoids duplication of work because of race conditions and is
+    // also fast because locked rows are ignored rather than blocking.
+    // It's only available in MySQL and PostgreSQL
+    if (['mysql', 'pg'].includes(tx.client.config.client)) {
```

Comment:
we'll most likely be using `mysql2` - should probably add that to the list if we're considering it at this point